types/mongoose/index.d.ts

```diff
@@ -3590,4 +3591,31 @@ declare module "mongoose" {
     /** The collection name */
     name: string;
   }
+
+  /* Helper type to extract a definition type from a Document type */
+  export type DocumentDefinition<T> = T extends Document & infer D ? D : never;
```

Comment:
Same comment as in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42652.

Reply:
Refactored this implementation with an `Omit` type.